### ISSUE TITLE
Slow download on app but no on website?

### ISSUE THREAD

**Issue by sprite-1** [itchio/itch#2337]
Whenever I download stuff using the itch.io app, the download speed is pretty low

![image](https://user-images.githubusercontent.com/33685433/67761850-11ec4c00-fa7f-11e9-96f7-1274a92dad13.png)

But when I download using the website, it downloads faster

![image](https://user-images.githubusercontent.com/33685433/67761962-48c26200-fa7f-11e9-9731-a4a11e1f82e1.png)

I monitored the processes and it doesn't seem like Windows Defender is interfering at all with it

What could be causing this?

**Comment by fasterthanlime** [comment]
Looks good from here

![image](https://user-images.githubusercontent.com/7998310/67786551-5bcd3600-fa6f-11e9-8660-bf5c01485324.png)

> I monitored the processes and it doesn't seem like Windows Defender is interfering at all with it

What in particular did you look for?

Are you downloading to an SSD or an HDD?

itch is extracting *while it's downloading the files*, and Windows is notoriously slow at creating lots of small files, so that might have something to do with it. The speed difference here is definitely not what I'd expect.

**Comment by sprite-1** [comment]
> What in particular did you look for?

Whenever Windows Defender decides to scan stuff, `MsMpEng.exe` starts consuming some CPU percentage (it happens sometimes when I compile stuff) but while downloading, it wasn't acting up.

> itch is extracting while it's downloading the files, and Windows is notoriously slow at creating lots of small files, so that might have something to do with it. The speed difference here is definitely not what I'd expect.

Is there a way to disable this behavior to check if that's what is slowing it down? I'm on an HDD

**Comment by fasterthanlime** [comment]
> Is there a way to disable this behavior to check if that's what is slowing it down? I'm on an HDD

Not exactly, but if you download a game with a .7z upload it'll do it in two passes: first download, then extract (that's because .7z files contains multiple files per compressed block, making enumerating files and pausing/resuming too expensive)

**Comment by fasterthanlime** [comment]
Woops, forgot the link: here's a . 7z game for you to test https://twoglasshams.itch.io/ashi-wash-classic (just checked it with Windows Defender, seems clean - also, a classic!)

**Comment by sprite-1** [comment]
> Woops, forgot the link: here's a . 7z game for you to test

Hmm... Apparently it's still exhibiting the same problem

![image](https://user-images.githubusercontent.com/33685433/67874296-b057da80-fb6f-11e9-9c60-a9c2c12d0a7e.png)

**Comment by fasterthanlime** [comment]
Can you try to download https://broth.itch.ovh/butler/windows-amd64-head/aeb1991a66e9f6cc10fd83cdf8535f56c2830fb3/archive/default extract it and run `butler diag --speed` in a Command Prompt.

Here are my results:

![image](https://user-images.githubusercontent.com/7998310/67905837-07ab7a00-fb73-11e9-9737-5f879aa950fb.png)

I'm hoping one of the last 4 line shows a significantly different speed for you - if not, I guess I'd be out of ideas for now.

I'm also interested in knowing:

  * What https://itch.io/country shows you
  * What your ISP is
  * If you currently have or ever had a proxy set up
  * What program you were using for the screenshot that showed 2000KB/s
  * What your version of Windows is

**Comment by sprite-1** [comment]
```
∙ Running diagnostics...
            CDN reachable | 366.0248ms
            Web reachable | 647.3726ms
            API reachable | 640.6711ms
          Broth reachable | 362.6046ms
         go https request | 20.50 MiB/s
                 eos Copy | 2.05 MiB/s
       eos Copy (to disk) | 1.85 MiB/s
    eos Extract (to disk) | 18.01 MiB/s
√ Everything went fine!
```

The other screenshot of the download is from a browser (Opera GX in this case) and it is on Windows 10 64-bit

![image](https://user-images.githubusercontent.com/33685433/67925950-ba212280-fbef-11e9-908d-270ad829b92a.png)

Also I hope you don't mind but I generally do not post personal information (such as location) publicly. 🙁 From what I can gleam from the stats though, it seems to be something related to a local operation

**Comment by fasterthanlime** [comment]
If you run diag again, do you get similar results? (Fast, slow, slow, fast?) In other words is that speed distribution reproducible or accidental?

**Comment by sprite-1** [comment]
These are three results from running it:
```
∙ Running diagnostics...
            CDN reachable | 357.821ms
            Web reachable | 637.4715ms
            API reachable | 749.5781ms
          Broth reachable | 320.0131ms
         go https request | 20.03 MiB/s
                 eos Copy | 173.44 KiB/s
       eos Copy (to disk) | 160.57 KiB/s
    eos Extract (to disk) | 15.50 MiB/s
√ Everything went fine!
```
```
∙ Running diagnostics...
            CDN reachable | 388.5065ms
            Web reachable | 780.0197ms
            API reachable | 638.5645ms
          Broth reachable | 287.8073ms
         go https request | 20.02 MiB/s
                 eos Copy | 2.13 MiB/s
       eos Copy (to disk) | 1.93 MiB/s
    eos Extract (to disk) | 17.18 MiB/s
√ Everything went fine!
```
```
∙ Running diagnostics...
            CDN reachable | 345.8443ms
            Web reachable | 650.8474ms
            API reachable | 657.5161ms
          Broth reachable | 344.4594ms
         go https request | 20.03 MiB/s
                 eos Copy | 2.25 MiB/s
       eos Copy (to disk) | 2.11 MiB/s
    eos Extract (to disk) | 18.29 MiB/s
√ Everything went fine!
```
On every attempt, it starts stalling at `eos Copy` so I leave it for a while and when I check back after a while it's done so I don't have an estimate how long it takes for `eos Copy` to finish.

**Comment by Rodancoci** [comment]
Hi, I am also experiencing low performance with eos Copy operations (whatever those are). Just ran `butler diag --speed` and here are my results:

```
• Running diagnostics...
            CDN reachable | 465.141622ms
            Web reachable | 720.499951ms
            API reachable | 717.640918ms
          Broth reachable | 845.925758ms
         go https request | 20.01 MiB/s
                 eos Copy | 7.46 MiB/s
       eos Copy (to disk) | 1.07 MiB/s
    eos Extract (to disk) | 20.14 MiB/s
✓ Everything went fine!
```

I'm running Arch Linux kernel 5.11.13-arch1-1 and have butler 15.20.0-1 installed.

**Comment by leafo** [comment]
Closing out old issues. Please leave a comment if you are still having trouble.
